engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java

```diff
@@ -1130,9 +1153,22 @@ IR.Expression translateDecimalLiteral(
   }
 
   IR.Literal translateLiteral(Tree.TextLiteral txt) {
-    var fullTxt = txt.codeRepr().trim();
-    var t = fullTxt.substring(1, fullTxt.length() - 1);
-    return new IR$Literal$Text(t, getIdentifiedLocation(txt), meta(), diag());
+    StringBuilder sb = new StringBuilder();
+    for (var t : txt.getElements()) {
+      var sectionTxt = switch (t) {
+          case TextElement.Section s -> s.getText().codeRepr();
+          case TextElement.Escape e -> switch (e.getToken().codeRepr()) {
+            case "\\n" -> "\n";
+            case "\\t" -> "\t";
+            case "\\r" -> "\r";
+            case "\\f" -> "\f";
+            default -> throw new UnhandledEntity(e, "translateLiteral");
+          };
```

Comment:
`e.getValue()` returns the value, which is a Unicode codepoint (or -1 if it was invalid). You can append it to the `StringBuilder` with `sb.appendCodePoint()`

Reply:
Used in e7b9b501a2ff6640086a86a2d09f8f8ce0d86e53 - thanks for the tip.